### PR TITLE
Clean python 2 artifacts. Fix #826

### LINKED ISSUE #826: Python 2.x artifacts present in code
`super(..., self).` instead of `super()`
subclassing of `object`
`IOError` instead of `OSError`

### PR DESCRIPTION
**Pull request by penguinolog**
* `IOError` is `OSError` in python3
* `io.open` -> `open`
* `return str()` -> `return ''` directly
* direct `super()` call
* do not subclass `object`

### CODE REVIEW

File: ipykernel/zmqshell.py [hunk 1]
```diff
@@ -378,7 +378,7 @@ def connect_info(self, arg_s):
         print ("Paste the above JSON into a file, and connect with:\n"
             "    $> jupyter <app> --existing <file>\n"
             "or, if you are local, you can connect with just:\n"
-            "    $> jupyter <app> --existing {0}\n"
+            "    $> jupyter <app> --existing {}\n"
```
Comment (davidbrochart) [inline ipykernel/zmqshell.py:381]: Maybe replace `.format(connection_file)` with:
```suggestion
            f"    $> jupyter <app> --existing {connection_file}\n"
```

**Review by davidbrochart — APPROVED** [review]
You also made the following changes:
- replace `.format()` with `f-strings`
- replace `set([1, 2])` with `{1, 2}`

File: ipykernel/zmqshell.py [hunk 2]
```diff
@@ -378,7 +378,7 @@ def connect_info(self, arg_s):
         print ("Paste the above JSON into a file, and connect with:\n"
             "    $> jupyter <app> --existing <file>\n"
             "or, if you are local, you can connect with just:\n"
-            "    $> jupyter <app> --existing {0}\n"
+            f"    $> jupyter <app> --existing {connection_file}\n"
             "or even just:\n"
             "    $> jupyter <app> --existing\n"
             "if this is the most recent Jupyter kernel you have started.".format(
```
Comment (davidbrochart) [inline ipykernel/zmqshell.py:384]: You also need to remove `.format(connection_file)`.

Comment (penguinolog) [reply ipykernel/zmqshell.py]: it was removed, like almost all other format calls in next commit

Comment (davidbrochart) [reply ipykernel/zmqshell.py]: Thanks!

**Pull request merged**